### PR TITLE
CC-11610 init commit

### PR DESCRIPTION
**Pull request by pyvovarov-s**
- Developer(s): @pyvovarov-s

- Ticket: https://spryker.atlassian.net/browse/CC-10508

- Release Group: https://release.spryker.com/release-groups/view/3087

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   ChecksumGenerator  | major (new) |                       |

-----------------------------------------

##### Change log

Initial Release
- Introduced data checksum generation functionality.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "spryker/checksum-generator",
+  "type": "library",
+  "description": "ChecksumGenerator module",
```
Comment (dereuromark) [inline composer.json]: If this is not supposed to be a normal core module, we must remove that word here.

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "spryker/checksum-generator",
+  "type": "library",
+  "description": "ChecksumGenerator",
+  "license": "proprietary",
```
Comment (abitskil) [inline composer.json]: Change the license to MIT.

File: .license [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/**
```
Comment (abitskil) [inline .license:1]: Replace the content of the file, use `vendor/spryker/decimal-object/.license` as a source.

File: LICENSE [hunk 4]
```diff
@@ -0,0 +1,40 @@
+SPRYKER SYSTEMS GMBH EVALUATION LICENSE AGREEMENT
```
Comment (abitskil) [inline LICENSE]: Replace the content of the file, use `vendor/spryker/decimal-object/LICENSE` as a source.

File: tests/SprykerTest/ChecksumGenerator/_support/ChecksumGeneratorTester.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+ * @SuppressWarnings(PHPMD)
+ */
+class ChecksumGeneratorTester extends Actor
+{
+    use _generated\ChecksumGeneratorTesterActions;
+
+    /**
+     * @return \Spryker\ChecksumGenerator\Checksum\ChecksumGeneratorInterface
+     */
+    public function getCrcProductConfigurationDataChecksumGenerator(): ChecksumGeneratorInterface
```
Comment (abitskil) [inline tests/SprykerTest/ChecksumGenerator/_support/ChecksumGeneratorTester.php]: Change the name to the `getCrcOpenSslChecksumGenerator()`.

File: tests/SprykerTest/ChecksumGenerator/ChecksumGeneratorTest.php [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+    /**
+     * @var \SprykerTest\ChecksumGenerator\ChecksumGeneratorTester
+     */
+    protected $tester;
+
+    /**
+     * @return void
+     */
+    public function testGenerateProductConfigurationDataChecksumGeneratesEncodedCheckSum(): void
```
Comment (abitskil) [inline tests/SprykerTest/ChecksumGenerator/ChecksumGeneratorTest.php]: Please fix the starting name for all the test cases to `testGenerateChecksum*`,

File: tests/SprykerTest/ChecksumGenerator/ChecksumGeneratorTest.php [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    {
+        // Arrange
+        $productConfiguration = [
```
Comment (abitskil) [inline tests/SprykerTest/ChecksumGenerator/ChecksumGeneratorTest.php]: Rename to `configuration` or `data`, check for the same places.

**Review by tamasnyulas — APPROVED** [review]

**Pull request merged**